whiteout/resolver.go

```diff
@@ -0,0 +1,94 @@
+package whiteout
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/indexer"
+)
+
+var (
+	_ indexer.Resolver = (*Resolver)(nil)
+)
+
+type Resolver struct{}
+
+func (r *Resolver) Resolve(ctx context.Context, ir *claircore.IndexReport, layers []*claircore.Layer) *claircore.IndexReport {
+	// Here we need to check if any of the packages
+	// found are moot due to whiteouts.
+	ls := newLayerSorter(layers)
+	finalPackages := map[string]*claircore.Package{}
+	finalEnvironments := map[string][]*claircore.Environment{}
+	for pkgID, pkg := range ir.Packages {
+		packageDeleted := false
+		// First env.IntroducedIn?
+		packageLayer := ir.Environments[pkgID][0].IntroducedIn.String()
+		for fileLayer, f := range ir.Files {
+			// Check if it's a whiteout file, if it applies to the package's
+			// file and if the layer the whiteout file came from came after.
```

Comment:
can we add s comment on why we don't look at the same layer (maybe even just say see the spec for why we don't bother)?